server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -451,6 +456,30 @@ private void setMailAttribute(Message message, Mail mail, String name) {
         }
     }
 
+    private void metricForDecrement() {
+        try {
+            dequeuedMailsMetric.decrement();
+            gaugeRegistry.register("mailQueueSize:" + queueName, queueSizeGauge());
```

Comment:
The entire concept of a gauge is to be registered once, in the constructor....

Note that this makes this method useless.